agent/consul/config_endpoint.go

```diff
@@ -260,6 +266,26 @@ func (c *ConfigEntry) List(args *structs.ConfigEntryQuery, reply *structs.Indexe
 			reply.Kind = args.Kind
 			reply.Index = index
 			reply.Entries = filteredEntries
+
+			// Generate a hash of the content driving this response. Use it to
+			// determine if the response is identical to a prior wakeup.
+			newHash, err := hashstructure_v2.Hash(filteredEntries, hashstructure_v2.FormatV2, nil)
```

Comment:
We could hash the modify indexes (and names) of the config entries just fine for all of the endpoints here except the intentions ones.

Handling the intentions ones is much harder because there's an extra layer of indirection involved between the config entries that originally held the line item intentions, and the results you got back from the state store function.

The thing I was noticing doing it this way for all situations is that it's a reusable pattern without a lot of extra overhead.

Reply:
Added a quick benchmark of hashing some simple config entries on one of the endpoints.